tests/test_ble_lifecycle_receive_targets.py

```
@@ -665,13 +665,244 @@ def join(timeout: float | None = None) -> None:
+        controller._close_after_fatal_read()
+        assert close_calls == []
+
+        iface._get_lifecycle_controller = lambda: None
+        controller._close_after_fatal_read()
+        assert close_calls == ["iface-close"]
+    finally:
+        iface._get_lifecycle_controller = original_get_lifecycle_controller
+        original_close()
+
+
+def test_receive_controller_fallback_state_and_override_branches(
```

Ruff (0.15.6)

[warning] Too many statements (57 > 50)

(PLR0915)